### PR TITLE
[addon] add service.autoexec add-on

### PR DESCRIPTION
**Pull request by DaVukovic**
## Description

As spoken about...
If https://github.com/xbmc/xbmc/pull/18356 gets merged this could be an alternative.

Let me know if anything needs to be changed. 
## Motivation and Context

The autoexec.py is used by many users in multiple ways and this makes it easier for ppl who don't want to write service add-ons.

This can be improved in many ways
- different filename
- different location of the script to be executed
and so on

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: addons/service.autoexec/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# service.autoexec
+
+A service add-on which allows or disallows the execution of autoexec.py
```
Comment (phunkyfish) [inline addons/service.autoexec/README.md:3]: Would make sense to add a paragraph on where autoexec.py is to be located. I.e.:

For systems without profiles configured the file should be located in the root of the `userdata` folder. In the case of profiles it should be located in the root `userdata/profilename`.

File: system/addon-manifest.xml [hunk 2]
```diff
@@ -54,5 +54,6 @@
   <addon>xbmc.metadata</addon>
   <addon>xbmc.python</addon>
   <addon>xbmc.webinterface</addon>
+  <addon>service.autoexec</addon>
```
Comment (phunkyfish) [inline system/addon-manifest.xml:57]: I think it needs to be removed from here it is to be disabled by default if I understood @garbear correctly?

Comment (garbear) [reply system/addon-manifest.xml:57]: Running autoexec.py, if present, should certainly be disabled by default. One option is to omit the add-on from the manifest. Instead, the approach taken in this PR is to ship an enabled add-on, with a setting that defaults to false.

Comment (phunkyfish) [reply system/addon-manifest.xml:57]: Ah, ok. Got it. Is there any benefit for this versus the other approach?

Comment (fuzzard) [reply system/addon-manifest.xml:57]: Imo the add-on should be default disabled, and personally I'd take it further and not have it installed. Available in repo, user then chooses to install.

**Comment by ronie** [comment]
appreciate the effort, but this wouldn't be my preferred way to handle the removal of autoexec.py support

1. you have created an addon that executes code outside of it's own root directory.
we are currently focusing on making our addon system more secure, with the ultimate goal to sandbox addons.
i don't really think this addon is the best example security wise and it may not survive if/once we get sandboxing working.

2. this is deprecated and will lead to a warning in the log: `RunScript(special://profile/autoexec.py)`
`WARNING <general>: CPythonInvoker(2): Script invoked without an addon. Adding all addon modules installed to python path as fallback. This behavior will be removed in future version.`

3. if you've created a autoexec.py that keeps running in the background, kodi can't properly stop the script when you quit kodi.
instead it has to hard-kill the script:
`ERROR <general>: CPythonInvoker(2, special://profile/autoexec.py): script didn't stop in 5 seconds - let's kill it`
if you're lucky, the script will crash at this point. but worst case scenario, it will hang kodi. 


imo if a user wishing to run some python code on startup, they should create a service addon for it.
a proper service addon doesn't suffer from all of the issues mentioned above.

**Comment by garbear** [comment]
The biggest downside of the service add-on, from a user perspective, is the add-on fluff. What's the best way to manage this so that we can keep the benefits of a service add-on?

**Comment by ronie** [comment]
we could update the https://kodi.wiki/view/Autoexec.py page with this info:

step 1. create a 'service.autoexec' folder in .kodi/addons
step 2. move your existing autoexec.py file to this folder
step 3. create an addon.xml file in this folder (example code: https://kodi.wiki/view/Service_add-ons)
step 4. enable your addon in kodi

**Comment by phunkyfish** [comment]
Is there a way to execute a script and if it’s still running on kodi exit, kill it?

Similar to having a process ID on Linux?

Having it run a script from it’s own addon dir is a good call. But it’s number 3 that is the most difficult to solve for.

**Comment by DaVukovic** [comment]
ok, too many flaws then

>step 1. create a 'service.autoexec' folder in .kodi/addons

Just keep in mind that this ^^ will kill the flexibillity for users who have multiple profiles which was also covered by this pr. 

Users can't have different add-ons for different profiles with different contents for the executable python file in use. Add-ons which are installed on the master profile are disabled by default for any other profile. Enabling them does work, but then the autoexec.py file in question (or any other service.py, default.py file)  has the very same content as it will have for the master profile. So users do have to write other checks in that autoexec.py file to know which profile is in use and act accordingly. Thus it will make the workflow way more complex and user-unfriendly as it was before we decided to drop autoexec.py support. 

As I already said at #18356 this is not an improvement. It's a step backwards. I understand the security approach mentioned here and that we should sandbox add-ons. That's completely fine. But then I would still suggest a Kodi GUI setting on the "standard"-level (like we do it for the webserver/webinterface) which gives the option to allow/disallow the execution of autoexec.py. Kodi GUI settings are seperated for each profile as every profile has its own guisettings.xml.

It was asked how a Kodi GUI setting is any different from a service add-on. Now I guess we know ;)

**Pull request closed without merging**